projects/out_of_sample/scripts/experiment-config/prep_oos_experiments.py

```diff
@@ -0,0 +1,168 @@
+
+
+def make_argo_submit_command(
+    bucket, project, experiment, trial: int, prognostic_run_config_path, segment_count
+) -> str:
+    trial_string = f"trial{trial}"
+    tag = f"{experiment}-{trial}"
+    job_name = f"prognostic-run-{uuid.uuid4().hex}"
+    argo_submit_str = (
+        "argo submit --from workflowtemplate/prognostic-run \\\n"
+        + f"-p bucket={bucket} \\\n"
+        + f"-p project={project} \\\n"
+        + f"-p tag={tag} \\\n"
+        + f"-p config='$(< {prognostic_run_config_path})' \\\n"
+        + f"-p segment-count={segment_count} \\\n"
+        + "-p memory='23Gi' \\\n"
+        + "-p cpu='24' \\\n"
+        + "-p online-diags-flags='--verification  1yr_pire_postspinup --n-jobs 5' \\\n"
+        + f"--name '{job_name}' \\\n"
+        + f"--labels 'project={project},experiment={experiment},"
+        + f"trial={trial_string}' \n\n"
+    )
+    output_path = (
+        f"gs://{bucket}/{project}/TODAY'S_DATE/"
+        + f"{experiment}-{trial_string}/fv3gfs_run"
+    )
+    return argo_submit_str, job_name, output_path
+
+
+def _cleanup_temp_dir(temp_dir):
+    temp_dir.cleanup()
+
+
+def prep_oos_experiments(args):
+    temp_dir = tempfile.TemporaryDirectory()
+    atexit.register(_cleanup_temp_dir, temp_dir)
+
+    with open(args.config_path, "r") as f:
+        try:
+            all_experiment_config = yaml.safe_load(f)
+        except yaml.YAMLError as exc:
+            print(exc)
+
+    prognostic_run_config_template_path = all_experiment_config[
+        "prognostic_run_config_template"
+    ]
+    launch_destination = all_experiment_config["launch_destination"]
+    if launch_destination.startswith("~"):
+        launch_destination = os.path.expanduser(launch_destination)
+    model_config_dir_path = all_experiment_config["model_config_dir_path"]
+
+    segments = all_experiment_config["segments"]
+
+    result_destination_config = all_experiment_config["result_destination"]
+    bucket = result_destination_config["bucket"]
+    project = result_destination_config["project"]
+    experiment_base = result_destination_config["experiment"]
+    trial = result_destination_config["trial"]
+
+    tendency_paths = {
+        "tq": all_experiment_config["base_model"]["tq_tendencies_path"],
+        "wind": all_experiment_config["base_model"]["wind_tendencies_path"],
+    }
+    n = 0
+    run_str = "#!/bin/bash\n\nset -e\n\n"
+    for novelty_detector in all_experiment_config["novelty_detectors"]:
+        nd_name = novelty_detector["nd_name"]
+        nd_path = novelty_detector["nd_path"]
+        for cutoff in novelty_detector["cutoffs"]:
+            n += 1
+            model_config_paths = []
+            for tendency, tendency_path in tendency_paths.items():
+                tendency_id = f"{nd_name}-{cutoff}-{tendency}"
+                model_config_path = os.path.join(model_config_dir_path, tendency_id)
+                temp_config_path = os.path.join(temp_dir.name, tendency_id)
```

Comment:
FYI If the temporary directory only needs to be referenced in one bit of the code rather than throughout, you can just use it within the context 
```
with tempfile.TemporaryDirectory() as tmpdirname:
    do stuff with the tmpdir and it gets deleted after this block
```
so you don't have to manually call cleanup or do the atexit stuff